### DISCUSSION TITLE
Support cdk diff output that can be understood by many markdown renderers

### DISCUSSION THREAD

**Discussion by sgmorrison** [aws/aws-cdk#19606] [General]
The output of a `cdk diff` command produces a diff summary like below:

```
Stack stack-name
Resources
[~] AWS::CloudWatch::Dashboard dashboard-name
 └─ [~] DashboardBody
     └─ [-] "old"
     └─ [+] "new"
```

When posting such a diff in a pull request to show expected changes, there is no good way to add colouring to the text. I propose we support a second output format which is compatible with the `diff` language support in markdown:

```diff
Stack stack-name
Resources
~ AWS::CloudWatch::Dashboard dashboard-name
~ └─ DashboardBody
-    └─ "old"
+    └─ "new"
```

This conveys the same information as before, but also assists in assessing planned changes.

The new format would need to be enabled with a command line flag, to retain backwards compatibility.

**Comment by indrora** [comment]
This would make for a good feature suggestion, but it would likely be a low priority in the long run.

**Comment by kaykhan** [comment]
@sgmorrison did you find a solution?

**Reply by sgmorrison** [reply]
Unfortunately not. I changed tactics instead, using snapshot tests with the snapshot checked in to version control. That way the diff is presented alongside code change diffs, and is highlighted well.

**Comment by gibbona1** [comment]
@sgmorrison @indrora do you know of any such feature that's been created since?

It'd be really nice to have something like what's mentioned above
